### ISSUE TITLE
Remove dependency on surrial in favor of JSON.stringify

### ISSUE THREAD

**Issue by nicojs** [stryker-mutator/stryker-js#2814]
**Is your maintenance request related to a problem? Please describe.**

We currently have a (probably unused) feature of being able to serialize regexes, maps, sets, and even _functions_ in the communication between the main process and worker processes. This was needed in the Stryker <4, because we wanted to send actual instances of `File` across the wire. I actually started a side project called [surrial](https://github.com/nicojs/surrial) for this purpose.

However, since we now moved away from the need of serializing files themselves, we should be able to migrate away from rolling our own serializing.

We should probably validate that no non-serializable values are present on the provided options. For example:

```js
// stryker.conf.js
module.exports = { 
  karma: {
    webpack: {
      some: /regex/ // 👈 this would be send as `null` to the worker process.
    }
  }
};
```

Maybe we could give an error message like:

> Provided option "karma.webpack.some" is not JSON serializable. This value is not allowed.

**Describe the solution you'd like**
Use plain old JSON.stringify to send objects across the wire.

**Describe alternatives you've considered**
We could also use the widely popular [serialize-javascript](https://github.com/yahoo/serialize-javascript). That way we would keep most of the features.

**Additional context**
This would be a breaking change since values like `RegEx`, `Set`, `Map`, etc in the config wouldn't be supported anymore.

**Comment by nicojs** [comment]
@kmdrGroch are you still planning to pick this one up? Otherwise, I can have a look.

**Comment by nicojs** [comment]
Closed with #2877
